Lombiq.HelpfulLibraries/Libraries/Mvc/RouteModel.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using OrchardCore.Admin;
+using OrchardCore.Environment.Extensions;
+using OrchardCore.Modules.Manifest;
+using OrchardCore.Mvc.Core.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Reflection;
+
+namespace Lombiq.HelpfulLibraries.Libraries.Mvc
+{
+    public class RouteModel
+    {
+        private readonly List<KeyValuePair<string, string>> _arguments;
+
+        private readonly Lazy<bool> _isAdminLazy;
+        private readonly Lazy<string> _routeLazy;
+
+        public RouteModel(

```

Comment:
Can't other OSS modules of ours use this right away?

Reply:
Sure. What do you have in mind? I've only looked for `@Url` and `GoToRelativeUri` usages within the OSOCE solution.

Reply:
Search for Razor TagHelpers that can use routes too.

Reply:
There are some in the training demo, but I thought that's for teaching about vanilla OC so it wouldn't be appropriate to replace them. Should I do it anyway?

Reply:
I have looked for asp-action and asp-controller. I found nothing else.

Reply:
OK then.